src/main/java/net/minestom/server/network/NetworkBufferTypeImpl.java

```diff
@@ -631,7 +631,8 @@ record RegistryTypeType<T extends ProtocolObject>(@NotNull Function<Registries,
         public void write(@NotNull NetworkBuffer buffer, DynamicRegistry.Key<T> value) {
             Check.stateCondition(buffer.registries == null, "Buffer does not have registries");
             final DynamicRegistry<T> registry = selector.apply(buffer.registries);
-            final int id = registry.getId(value);
+            // For whatever reason, the client indexes painting variants starting at 1, so we add 1 in that case.
+            final int id = registry.id().equals("minecraft:painting_variant")?registry.getId(value)+1:registry.getId(value);
```

Comment:
Its not random, its because painting variants are encoded as a (mojang term) "holder", which means either its a painting id from the registry or you can send the entire painting variant data directly in which case you write an id of 0 and the data following it.

We don't have proper support for these holder types for now so I am ok leaving this with a fixme. Something like:
```
// Painting variants may be sent in their entirety rather than a registry reference so the ID is offset by 1 to indicate this.
// FIXME: Support sending the entire registry object instead of an ID reference.
final int id = registry.id().equals("minecraft:painting_variant") ? registry.getId(value) + 1 : registry.getId(value);
```